src/libs/sfdk/dockervirtualmachine.cpp

```diff
@@ -244,17 +241,91 @@ void DockerVirtualMachinePrivate::start(const QObject *context, const Functor<bo
     sharePath(Constants::BUILD_ENGINE_SHARED_CONFIG_MOUNT_POINT, cachedInfo().sharedConfig);
     sharePath(Constants::BUILD_ENGINE_SHARED_SSH_MOUNT_POINT, cachedInfo().sharedSsh);
 
+    arguments.append("--name");
     arguments.append(q->name());
 
-    auto runner = std::make_unique<DockerRunner>(arguments);
-    QObject::connect(runner.get(), &DockerRunner::done, q,
-            [this, context, functor, process = runner->process()](bool ok) {
-        if (ok)
-            containerId = QString::fromLocal8Bit(process->readAllStandardOutput().trimmed());
-        if (context)
-            functor(ok);
+    arguments.append(q->name());
+
+    return arguments;
+}
+
+void DockerVirtualMachinePrivate::start(const QObject *context, const Functor<bool> &functor)
+{
+    Q_Q(DockerVirtualMachine);
+    Q_ASSERT(context);
+    Q_ASSERT(functor);
+
+    const QPointer<const QObject> context_{context};
+
+    auto prepare = [=](const QStringList &arguments, CommandQueue::BatchId batch) {
+        auto runner = std::make_unique<DockerRunner>(arguments);
+        QObject::connect(runner.get(), &DockerRunner::failure, Sdk::instance(), [=]() {
+            commandQueue()->cancelBatch(batch);
+            callIf(context_, functor, false);
+        });
+#ifdef Q_OS_MACOS
+        return std::move(runner);
+#else
+        return runner;
+#endif
+    };
+
+    auto enqueue = [=](const QStringList &arguments, CommandQueue::BatchId batch) {
+        auto runner = prepare(arguments, batch);
+        DockerRunner *runner_ = runner.get();
+        commandQueue()->enqueue(std::move(runner));
+        return runner_;
+    };
+
+    auto enqueueImmediate = [=](const QStringList &arguments, CommandQueue::BatchId batch) {
+        auto runner = prepare(arguments, batch);
+        DockerRunner *runner_ = runner.get();
+        commandQueue()->enqueueImmediate(std::move(runner));
+        return runner_;
+    };
+
+    CommandQueue::BatchId batch = commandQueue()->beginBatch();
+
+    QStringList psArguments;
```

Comment:
The prefixes here exist just to deal with name conflicts. If it was the only arguments list in the scope, it would be named just "arguments", and so I do not follow the convention with long names here. But now I noticed it should be 'ls'.

Reply:
Renamed using 'ls' prefix

Reply:
I would still prefer `containerLsArguments`, but I can live with `ls` :)